### PR TITLE
Add the `Config::getCurrentServer()` method 

### PR DESCRIPTION
**Pull request by MauricioFauth**
The `Config::getCurrentServer()` method returns the current server configuration.
It's a value object that is equivalent to `$GLOBALS['cfg']['Server']` setting.

### CODE REVIEW

File: libraries/classes/Config.php [hunk 1]
```diff
@@ -86,17 +86,14 @@ class Config
 
     public bool $errorConfigFile = false;
 
-    /** @var mixed[] */
-    public array $defaultServer;
-
     private bool $isHttps = false;
 
     private Settings $config;
+    private Settings\Server|null $currentServer = null;
```
Comment (kamil-tekiela) [inline libraries/classes/Config.php:92]: Could this be imported?

Comment (MauricioFauth) [reply libraries/classes/Config.php]: @kamil-tekiela I was not happy with that property, so I removed it. Instead, I created a `$server` property that is equivalent to `$GLOBALS['server']` and updated the `getCurrentServer()` method.

File: libraries/classes/Config.php [hunk 2]
```diff
@@ -1252,4 +1241,9 @@ public function getSettings(): Settings
     {
         return $this->config;
     }
+
+    public function getCurrentServer(): Settings\Server|null
```
Comment (kamil-tekiela) [inline libraries/classes/Config.php:1245]: And what about this one?

Comment (MauricioFauth) [reply libraries/classes/Config.php:1245]: Do you mean like injecting a value for testing?
Something like that:
```php
$config = new Config();
$config->config = new Settings(['Servers' => [2 => ['host' => 'testhost']]]);
$config->server = 2;
$currentServer = $config->getCurrentServer();
```

Comment (kamil-tekiela) [reply libraries/classes/Config.php:1245]: No, sorry. I meant can we replace `Settings\Server` with just `Server`?

Comment (MauricioFauth) [reply libraries/classes/Config.php]: Haha of course. That makes sense. Done!

**Pull request merged**